### PR TITLE
global: removal of six, support for python -m dojson, testing PyPy

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

**Comment by kaplun** [comment]
To better understand: 
- what is the reason for removing `six`? Aren't we ending up reimplementing `six` into the `_compat.py`? 
- is this going to be part of an horizontal removal of six everywhere?

**Comment by jirikuncar** [comment]
@kaplun `six` is nice for quick compatibility, but if we are using only limited set of features it is a bit overkill. It's usually possible to write compatible PY2/3 code without using `six` or python version detection.

**Comment by tiborsimko** [comment]
The releases of `six` are [not very frequent](https://bitbucket.org/gutworth/six/downloads), which is good, but we may have to watch for any future e.g. security related updates, if some of our packages are going to use _a lot_ of `six`.  Otherwise we should be good.

Perhaps leave a comment in `_compat.py` saying the code was inspired/borrowed from `six`? And, for Invenio standalone packages, we can perhaps centralise  some stuff in a common package? (`Invenio-Base`?)

File: dojson/_compat.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# This file is part of DoJSON
+# Copyright (C) 2016 CERN.
+#
+# DoJSON is free software; you can redistribute it and/or
+# modify it under the terms of the Revised BSD License; see LICENSE
+# file for more details.
+
+"""Compatibility module for Python 2 and 3.
+
+The code is inspired by ``six`` library and cheat sheet from
```
Comment (jirikuncar) [inline dojson/_compat.py:12]: ping @tiborsimko

**Pull request merged**